pkgs/by-name/ol/olm/package.nix

```
@@ -21,11 +21,18 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
-  postPatch = ''
-    substituteInPlace olm.pc.in \
-      --replace '$'{exec_prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
-      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
-  '';
+  postPatch =
+    ''
+      substituteInPlace olm.pc.in \
+        --replace '$'{exec_prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
+        --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+    ''
+    # Clang 19 has become more strict about assigning to const variables
+    # Patch from https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=281497
+    + lib.optionalString stdenv.cc.isClang ''
```

I guess, since it only applies to Clang 19 and above, this is a tad more correct?
```suggestion
    + lib.optionalString (stdenv.cc.isClang && lib.versionAtLeast clangStdenv.cc.version "19") ''
```

---

Thanks for the suggestion, got it! ZHF is such an interesting initiative and sometimes even a bit competitive, which makes it all the more fun!

---

Hahah as you might've seen, I also made a PR. I did check if someone else did but I saw nothing, I struggled for a bit to find the culprit (I don't own a darwin machine), went on godbolt, finally submitted a fix, found myselves beaten by half an hour 😆

---

Haha, looks like I was only faster because I happened to have a Darwin machine within arm’s reach!